feature - v21.03.0 update (#66)
dgraph-io/charts: helmfiles/tests/README.md

# Tests

> These tests Dgraph delployments with a variety helm config values.  

Reviewer: ```suggestion
These tests validate Dgraph deployments with a variety of helm config values.  
```

  --volume $PWD/dgraph_tls:/dgraph_tls \
  --name "dgraph-certs" \
  "dgraph/standalone:v21.03.0"

## Copy script to into container
docker cp \
  ../../charts/dgraph/scripts/make_tls_secrets.sh \
  dgraph-certs:/make_tls_secrets.sh

## Generate TLS certs per each test
TLS_TESTS="alpha-tls zero-tls"
for TEST in $TLS_TESTS; do
  docker exec -it dgraph-certs /make_tls_secrets.sh --release "test" \
    --client "dgraphuser" --namespace dgraph-test-$TEST --zero --tls_dir /dgraph_tls/dgraph-test-$TEST
done

## Verify certs/keys hostnames
for TEST in $TLS_TESTS; do
  echo "Dgraph Alpha: Release[test], Namespace [dgraph-test-$TEST]:"
  echo "-------------------------------------------------------------"  
  docker exec -it dgraph-certs dgraph cert ls --dir /dgraph_tls/dgraph-test-$TEST/alpha | awk -F: '/Hosts/{gsub(/\[ ]+/, "", $2); print $2}' | tr , '\n'
  echo "Dgraph Zero: Release[test], Namespace [dgraph-test-$TEST]:"
  echo "-------------------------------------------------------------"
  docker exec -it dgraph-certs dgraph cert ls --dir /dgraph_tls/dgraph-test-$TEST/zero | awk -F: '/Hosts/{gsub(/\[ ]+/, "", $2); print $2}' | tr , '\n'
  printf "\n"
done
```

## Run Tests

> This will deploy a dgraph cluster using different types of dgraph configuration.  Use the `--environment $TEST` flag to select the desired test.  

Reviewer: ```suggestion
This will deploy a Dgraph cluster using different types of Dgraph configuration.  Use the `--environment $TEST` flag to select the desired test.  
```

> The focus on these tests is to deploy Dgraph on Kubernetes, and verify that dgraph is running successful.

Reviewer: ```suggestion
The focus on these tests is to deploy Dgraph on Kubernetes, and verify that Dgraph is running successfully.
```
